src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,410 @@
+        {
+            var info = _bindings.Infos[iinfo];
+            TFDataType inputType = GetTFType(input.Schema.GetColumnName(_bindings.Infos[iinfo].SrcIndices[0]));
+            TFDataType outputType = GetTFType(_bindings.OutputColNames[iinfo]);
+            if ( (inputType  == TFDataType.Float || ((int)inputType) == 101) && outputType == TFDataType.Float)
+            {
+                Func<IRow, int, ValueGetter<VBuffer<float>>> del = MakeGetter<float, float>;
+                var meth = del.GetMethodInfo().GetGenericMethodDefinition().MakeGenericMethod(typeof(float), typeof(float));
+                return (Delegate)meth.Invoke(this, new object[] { input, iinfo });
+            }
+
+            if ( (inputType == TFDataType.Int32 || ((int)inputType) == 103) && outputType == TFDataType.Float)
+            {
+                Func<IRow, int, ValueGetter<VBuffer<float>>> del = MakeGetter<int, float>;
+                var meth = del.GetMethodInfo().GetGenericMethodDefinition().MakeGenericMethod(typeof(int), typeof(float));
+                return (Delegate)meth.Invoke(this, new object[] { input, iinfo });
+            }
+
+            if ((inputType == TFDataType.Float || ((int)inputType) == 101) && outputType == TFDataType.Int32)
+            {
+                Func<IRow, int, ValueGetter<VBuffer<int>>> del = MakeGetter<float, int>;
+                var meth = del.GetMethodInfo().GetGenericMethodDefinition().MakeGenericMethod(typeof(float), typeof(int));
+                return (Delegate)meth.Invoke(this, new object[] { input, iinfo });
+            }
+            if ((inputType == TFDataType.Int32 || ((int)inputType) == 103) && outputType == TFDataType.Int32)
+            {
+                Func<IRow, int, ValueGetter<VBuffer<int>>> del = MakeGetter<int, int>;
+                var meth = del.GetMethodInfo().GetGenericMethodDefinition().MakeGenericMethod(typeof(int), typeof(int));
+                return (Delegate)meth.Invoke(this, new object[] { input, iinfo });
+            }
+
+            return null;
+        }*/
+
+        private ValueGetter<VBuffer<float>> MakeGetter(IRow input, int iinfo)
+        {
+            Host.AssertValue(input);
+            // Need to check type here.
+            //Host.Assert(Infos[iinfo].TypeSrc.IsText);
+
+            var info = _bindings.Infos[iinfo];
+            var srcGetterOnes = new ValueGetter<float>[info.SrcIndices.Length];
+            var srcGetterVecs = new ValueGetter<VBuffer<float>>[info.SrcIndices.Length];
+            for (int j = 0; j < info.SrcIndices.Length; j++)
+            {
+                if (info.SrcTypes[j].IsVector)
+                    srcGetterVecs[j] = GetSrcGetter<VBuffer<float>>(input, iinfo, j);
+                else
+                    srcGetterOnes[j] = GetSrcGetter<float>(input, iinfo, j);
+            }
+
+            return (ref VBuffer<float> dst) =>
+                {
+                    var runner = _session.GetRunner();
```

Comment:
>runner [](start = 24, length = 6)

Can you have multiple runners at the same time from the same session?

It feels to me that the runner should be external to the getter delegate, and reused between calls. If this can not be done, explain why.

Reply:
It seems like we can not have single runner mostly because of implementation of `Runner` which currently does not allow to set the input value once it has been set (using `AddInput` method). This is the limitation of `TensorflowSharp`.

---
In reply to: [209305767](https://github.com/zeahmed/machinelearning/pull/3#discussion_r209305767) [](ancestors = 209305767)

Reply:
we will still need to think of what functionality can be extracted out of the delegate.
For example, `input.Schema.GetColumnName(info.SrcIndices[i])` is not going to change from call to call

---
In reply to: [209413407](https://github.com/zeahmed/machinelearning/pull/3#discussion_r209413407) [](ancestors = 209413407,209305767)

Reply:
You can and should have multiple Runners, the Runner isn't really a heavy object, it's just a holder for the state of a run.  As such it doesn't make much sense to try to share it between runs.

---
In reply to: [209663760](https://github.com/zeahmed/machinelearning/pull/3#discussion_r209663760) [](ancestors = 209663760,209413407,209305767)